addons/l10n_latam_check/models/account_payment.py

```diff
@@ -1,328 +1,250 @@
+                    if last_operation and last_operation[0].date > date:
+                        msgs.append(_(
+                            "It seems you're trying to move a check with a date (%s) prior to last operation done with "
+                            "the check (%s). This may be wrong, please double check it. By continue, the last operation on "
+                            "the check will remain being %s",
+                            format_date(self.env, date), last_operation.display_name, last_operation.display_name))
 
-    @api.onchange('l10n_latam_check_issuer_vat')
-    def _clean_l10n_latam_check_issuer_vat(self):
-        for rec in self.filtered(lambda x: x.l10n_latam_check_issuer_vat and x.company_id.country_id.code):
-            stdnum_vat = stdnum.util.get_cc_module(rec.company_id.country_id.code, 'vat')
-            if hasattr(stdnum_vat, 'compact'):
-                rec.l10n_latam_check_issuer_vat = stdnum_vat.compact(rec.l10n_latam_check_issuer_vat)
+        return msgs
 
-    @api.constrains('l10n_latam_check_issuer_vat', 'company_id')
-    def _check_l10n_latam_check_issuer_vat(self):
-        for rec in self.filtered(lambda x: x.l10n_latam_check_issuer_vat and x.company_id.country_id):
-            if not self.env['res.partner']._run_vat_test(rec.l10n_latam_check_issuer_vat, rec.company_id.country_id):
-                error_message = self.env['res.partner']._build_vat_error_message(
-                    rec.company_id.country_id.code.lower(), rec.l10n_latam_check_issuer_vat, 'Check Issuer VAT')
-                raise ValidationError(error_message)
+    def _get_reconciled_checks_error(self):
+        checks_reconciled = self.l10n_latam_check_ids.filtered(lambda x: x.issue_state in ['debited', 'voided'])
+        if checks_reconciled:
+            raise UserError(_(
+                "You can't cancel or re-open a payment with checks if some check has been debited or been voided. Checks:\n"
+                "%s") % ('\n'.join(['* %s (%s)' % (x.name, x.issue_state) for x in checks_reconciled])))
+
+    def action_cancel(self):
+        self._get_reconciled_checks_error()
+        super().action_cancel()
+
+    def action_draft(self):
+        self._get_reconciled_checks_error()
+        super().action_draft()
+
+    def _l10n_latam_check_split_move(self):
+        for payment in self.filtered(lambda x: x.payment_method_code == 'own_checks' and x.payment_type == 'outbound'):
+            if len(payment.l10n_latam_check_ids) == 1:
+                liquidity_line, counterpart_lines, dummy = payment._seek_for_lines()
+                payment.l10n_latam_check_ids.liquidity_line_id = liquidity_line.id
+                continue
+            move_id = self.env['account.move'].create({
+                'journal_id': payment.journal_id.id,
+                'move_type': 'entry',
+            })
+            liquidity_line, counterpart_lines, dummy = payment._seek_for_lines()
```

Comment:
For the record, now failing ci/style here

Reply:
Done